.github/workflows/main.yml

```diff
@@ -79,7 +79,7 @@ jobs:
         #   use-after-scope, double-free, invalid-free, and memory leaks.
         # * MemorySanitizer detects uninitialized reads.
         #
-        RUSTFLAGS: "-Zsanitizer=address"
+        RUSTFLAGS: "-Zsanitizer=address -C link-args=-lasan"
```

Comment:
Did you run into a problem that led you to add this?

I spent a lot of time debugging this one to hopefully get it right - I don't think this change is correct. I tried it this way at one point, but it can cause issues depending on the toolchain.

If everything is configured correctly (meaning all components regardless of language are built with `-fsanitize-address` or `-Zsanitizer=address`), `-lsan` is not required (assuming you're compiling with an llvm toolchain and not gcc). 

Forcibly linking against libsan can make it look like memory sanitizer is working but will cause problems - when the compiler is aware of asan (via correct setup) it ensures the symbols end up correctly using the asan versions. But merely linking against asan ensures that the asan symbols are present but doesn't guarantee the asan version of external calls will be used instead of the real version because you're not guaranteeing the final link order.

Reply:
Yes, the problem was that printf-compat was being compiled with ASAN but not linking against the ASAN libs, causing link errors. I'll check if this is still needed; if not I'll revert it.